server/user.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+var database = require('./database.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/** @const {RegExp} */
+user.VALID_EMAIL_REGEX =
```

Comment:
You are defining RegExp here so the utils function could be RegExp typed and does not need "RegExp(regex)"